CMakeLists.txt

```diff
@@ -678,6 +678,15 @@ ELSE(QMC_CUDA)
   MESSAGE(STATUS "Disabling CUDA")
 ENDIF(QMC_CUDA)
 
+IF(USE_NVTX)
+  IF(HAVE_CUDA)
+    ADD_DEFINITIONS(-DUSE_NVTX)
```

Comment:
I also prefer to avoid ADD_DEFINITIONS.
If a macro is not used local to a file, we have `src/config.h.cmake.in` to register all the macros.

Reply:
Moving to config.h makes sense.